services/web/server/src/simcore_service_webserver/folders/_folders_db.py

```diff
@@ -142,18 +149,18 @@ async def list_(
         list_query = base_query.order_by(desc(getattr(folders_v2.c, order_by.field)))
     list_query = list_query.offset(offset).limit(limit)
 
-    async with get_database_engine(app).acquire() as conn:
+    async with pass_or_acquire_connection(get_asyncpg_engine(app), connection) as conn:
         count_result = await conn.execute(count_query)
-        total_count = await count_result.scalar()
+        total_count = count_result.scalar()
```

Comment:
I think that all the `conn.exectute` can be changed with `conn.stream`

As you can see there is `scalar` on the `AsyncResult` object. https://docs.sqlalchemy.org/en/20/orm/extensions/asyncio.html#sqlalchemy.ext.asyncio.AsyncResult.scalar

My reasoning for stating this is the following: after you run the query via `execute/stream`, then you get a cursor. If you look at the docs (linked above) it said it is closing the cursor. So this means that you scalar operation is performed using a cursor which implies communication with the Database. Meaning and await should be used here, otherwise it could block the asyncio loop.


```suggestion
        count_result = await conn.stream(count_query)
        total_count = count_result.scalar()
```